app/src/androidTest/java/com/ifmo/rmp/MainScreenTest.kt

```
@@ -0,0 +1,139 @@
+        }
+
+        // Navigate back to Main screen
+        try {
+            composeTestRule.onNodeWithText("Home").performClick()
+            
+            composeTestRule.waitUntil(timeoutMillis = 5000) {
+                runCatching {
+                    composeTestRule.onNodeWithTag("profile_button").fetchSemanticsNode() != null
+                }.isSuccess
+            }
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)